### ISSUE TITLE
VM with data disks can't deploy to existing vnet/subnet

### ISSUE THREAD

**Issue by huckj** [test-kitchen/kitchen-azurerm#66]
I am able to successfully create a test-kitchen VM that targets an existing vnet/subnet.  I am also able to successfully create a test-kitchen VM that has data disks attached.  If i try to combine the scenarios and create a VM with data disks in an existing vnet/subnet, test-kitchen fails with the below exception.  It appears that in this combined scenario, the JSON for adding the data disks is being inserted into the "diagnosticsProfile" section.  I've tried both 0.12.1 and 0.12.0.  Any thoughts?

kitchen.yml:

```
platforms:
- name: windows2016
  driver_config:
    image_urn: MicrosoftWindowsServer:WindowsServer:2016-Datacenter:latest
    data_disks:
      - lun: 0
        disk_size_gb: 128
    vnet_id: /subscriptions/<SUBSCRIPTION_GUID>/resourceGroups/<RESOURCE_GROUP>/providers/Microsoft.Network/virtualNetworks/<VNET>
    subnet_id: <SUBNET>
```
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Failed to complete #create action: [409: unexpected token at '{
            "apiVersion": "2017-03-30",
            "type": "Microsoft.Compute/virtualMachines",
            "name": "[variables('vmName')]",
            "location": "[variables('location')]",
            "dependsOn": [
                "[concat('Microsoft.Network/networkInterfaces/', variables('nicName'))]"
            ],
            "properties": {
                "hardwareProfile": {
                    "vmSize": "[variables('vmSize')]"
                },
                "osProfile": {
                    "computername": "[variables('vmName')]",
                    "adminUsername": "[parameters('adminUsername')]",
                    "adminPassword": "[parameters('adminPassword')]"
                },
                "storageProfile": {
                    "imageReference": {
                        "publisher": "[parameters('imagePublisher')]",
                        "offer": "[parameters('imageOffer')]",
                        "sku": "[parameters('imageSku')]",
                        "version": "[parameters('imageVersion')]"
                    },
                    "osDisk": {
                        "name": "osdisk",
                        "createOption": "FromImage"
                    }
                },
                "networkProfile": {
                    "networkInterfaces": [
                        {
                            "id": "[resourceId('Microsoft.Network/networkInterfaces',variables('nicName'))]"
                        }
                    ]
                },
                "diagnosticsProfile": {
                      ,"dataDisks":
                          [{"name":"datadisk0","lun":0,"diskSizeGB":128,"createOption":"Empty"}]
                }
            },
            "tags": {
                "provider": "[variables('provider')]"

            }
        }
    ]
}
'] on default-windows2016-noformat
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration

**Comment by huckj** [comment]
Had a closer look at this and it appears that /templates/internal.erb is executed when deploying to an existing vnet/subnet, and that the code for adding adding data disks is in the "diagnosticsProfile" section (lines 330-333) and not the "storageProfile" section where it ought to be.  It's probably very straightforward, but I don't know my way around Ruby well enough to fork/fix it.  thought I'd at least point it out, though. :o)

**Comment by stuartpreston** [comment]
This should be fixed in Release 0.12.2 available on Rubygems.

**Comment by huckj** [comment]
Yes!  I grabbed the new gem yesterday and tested.  Thanks!
